src/osp/universe/universe.h

```diff
@@ -162,4 +127,58 @@ struct Universe
     std::vector<CoSpaceCommon>     m_coordCommon;
 };
 
+template <typename VEC_T, typename INDEX_T, typename ... RANGE_T>
+constexpr VEC_T to_vec(INDEX_T i, RANGE_T&& ... args) noexcept
+{
+    return {args[std::size_t(i)] ...};
+}
+
+
+/**
+ * @brief Get StridedArrayView1Ds from all components of a SatVectorDesc
+ *
+ * @param satVec   [in] SatVectorDesc describing layout in rData
+ * @param rData    [in] unsigned char* Satellite data, optionally const
+ * @param satCount [in] Range of valid satellites
+ *
+ * @return std::array of views, intended to work with structured bindings
+ */
+template <typename T, int DIMENSIONS_T, typename DATA_T, int ... COUNT_T>
```

Comment:
dimensions and count should be unsigned integer types, probably size_t.

signed vs unsigned is important because of the knock-on effects that it can have with integer overflow during multiplcation and all sorts of other crappy issues that are very hidden deep in implicit behavior land.